vowpalwabbit/learner.h

```diff
@@ -82,13 +93,33 @@ struct learn_data
   fn predict_f;
   fn update_f;
   multi_fn multipredict_f;
+
+  bool operator==(const learn_data& other) {
+    if (base != other.base) return false;
+    if (learn_f != other.learn_f) return false;
+    if (predict_f != other.predict_f) return false;
+    if (update_f != other.update_f) return false;
+    if (multipredict_f != other.multipredict_f) return false;
+    return true;
+  }
+  bool operator!=(const learn_data& other) {
+    return !(*this == other);
+  }
 };
 
 struct sensitivity_data
 {
   using fn = float (*)(void* data, base_learner& base, example& ex);
   void* data;
   fn sensitivity_f;
+
+  bool operator==(const sensitivity_data& other) {
+    if (sensitivity_f != other.sensitivity_f) return false;
+    return true;
```

Comment:
```suggestion
    return sensitivity_f == other.sensitivity_f;
```
For the others too. It removes the branching,

Reply:
I did this for consistency and readability. I imagine we'd switch all the equivalency operators to
`return x == other.x && y == other.y && z == other.z;` in the final implementation